### PR TITLE
Quote arguments containing spaces when launching OmniSharp.

### PR DESCRIPTION
**Pull request by JoeRobich**
Resolves https://github.com/OmniSharp/omnisharp-vscode/issues/5150

Tested locally with a Folder name containing a space. Failed to launch prior to this change and opens as expected with the change.

### CODE REVIEW

File: src/omnisharp/launcher.ts [hunk 1]
```diff
@@ -339,10 +339,16 @@ function getConfigurationValue(globalConfig: vscode.WorkspaceConfiguration, csha
 }
 
 async function launchDotnet(launchInfo: LaunchInfo, cwd: string, args: string[], platformInfo: PlatformInformation, options: Options, dotnetResolver: IHostExecutableResolver): Promise<LaunchResult> {
+    function escapeIfNeeded(arg: string) {
+        const hasSpaceWithoutQuotes = /^[^"].* .*[^"]/;
```
Comment (333fred) [inline src/omnisharp/launcher.ts:343]: What does this regex actually mean? A space not surrounded by quotes?

Comment (JoeRobich) [reply src/omnisharp/launcher.ts:343]: yeah stole it from the launchWindows method. Could just check startsWith('"') and contains(' ') instead.

Comment (333fred) [reply src/omnisharp/launcher.ts:343]: I am slightly worried about quotes in the middle of arguments: should I be? Or will that break things elsewhere?

Comment (333fred) [reply src/omnisharp/launcher.ts:343]: > Could just check startsWith('"')

Make sure you account for leading/trailing spaces that don't matter to the command.

**Comment by nohwnd** [comment]
https://github.com/nodejs/node/issues/7367#issuecomment-315210235 Recommends using cross-spawn package which implements this correctly without the need to manually escape, or at least that is what they say. Might be worth a try.

**Comment by nohwnd** [comment]
https://www.npmjs.com/package/cross-spawn

**Comment by JoeRobich** [comment]
Updated to use cross-spawn package. Need to verify Windows launch behavior.

**Review by 333fred — APPROVED** [review]
LGTM as long as we test the behavior.

**Comment by JoeRobich** [comment]
Confirmed works with spaces in workspace path on windows with both "omnisharp.useModernNet": "true" & "false"

**Pull request merged**